client/consensus/slots/src/lib.rs

```diff
@@ -242,6 +250,51 @@ pub trait SimpleSlotWorker<B: BlockT> {
 		Some(proposal)
 	}
 
+	/// Calls `create_inherent_data` and handles errors.
+	async fn create_inherent_data(
+		slot_info: &SlotInfo<B>,
+		logging_target: &str,
+	) -> Option<sp_inherents::InherentData> {
+		let now = Instant::now();
+		let remaining_duration = if now > slot_info.ends_at {
+			Duration::from_millis(0)
```

Comment:
nit: Good catch. But I'm wondering what is the behavior of `timeout` when the duration is `0`? Sounds like it might always error:

https://docs.rs/async-std/latest/async_std/prelude/trait.FutureExt.html#method.timeout

> Waits for both the future and a timeout, if the timeout completes before the future, it returns a TimeoutError.

If that is the case, then when `now > slot_info.ends_at` we can just return early.

Reply:
Changed to use delay. So code should pretty much return immediately anyway and saves us adding another `warn!()` block here, imho